spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/config/DiscoveryClientOptionalArgsConfiguration.java

```diff
@@ -204,4 +207,56 @@ static class OnJerseyClientDisabled {
 
 	}
 
+	static class RestTemplateEnabledCondition extends AnyNestedCondition {
+
+		RestTemplateEnabledCondition() {
+			super(ConfigurationPhase.REGISTER_BEAN);
+		}
+
+		@Conditional(JerseyClientNotPresentOrNotEnabledCondition.class)
+		static class OnJerseyClientNotPresentOrNotEnabled {
+
+		}
+
+		@ConditionalOnProperty(prefix = "eureka.client", name = "webclient.enabled", matchIfMissing = true,
+				havingValue = "false")
+		static class OnWebClientDisabled {
+
+		}
+
+		@ConditionalOnProperty(prefix = "eureka.client", name = "restclient.enabled", matchIfMissing = true,
+				havingValue = "false")
+		static class OnRestClientDisabled {
+
+		}
+	}
+
+	@ConditionalOnMissingClass("org.glassfish.jersey.client.JerseyClient")
+	@ConditionalOnClass(name = "org.springframework.web.client.RestClient")
+	@ConditionalOnProperty(prefix = "eureka.client", name = "restclient.enabled", havingValue = "true")
+	protected static class RestClientConfiguration {
```

Comment:
This should also have `@ConditionalOnProperty(prefix = "eureka.client", name = "webclient.enabled", matchIfMissing = true, havingValue = "false")` to allow the users to manually disable it in favour of WebClient (in a scenario where both are on the classpath).

Reply:
I added `RestClientEnabledCondition`.